### ISSUE TITLE
NPM destination plugin: failed to apply builtin javascript transform: json: unsupported type

### ISSUE THREAD

**Issue by lutek** [jitsucom/jitsu#965]
## Environment

- **Deploy method**: 
```
version: "2.3"
networks:
  web:
    external: true
services:
  jitsu:
    container_name: jitsu_server
    #image: jitsucom/server:latest
    image: jitsucom/server:1.42.7
    depends_on:
      - redis
    volumes:
      - ./data/jitsu/:/home/eventnative/data
    ports:
      - "8001:8001"
    networks:
      web:
        aliases:
          - jitsu
  redis:
    container_name: jitsu_redis
    image: redis:6.2.5-alpine
    volumes:
      - ./data/redis/:/data
    ports:
      - "32769:6379"
    networks:
      - web
```
- **Jitsu version**:  `1.42.7`
- **OS**:  `Mac OS BigSur` 

## Description
i wrote plugin (destination) according to docs, i made all test (passed), made exec (passed) but testing with jitsu server got error on processing event `failed to apply builtin javascript transform: json: unsupported type: map[interface {}]interface {}`

is this problem with plugin? how to find the problem?

```
{"event":{"_timestamp":"2022-09-07T08:34:44.466601Z","api_key":"js.e1af3113","available":true,"categories":["3bed7dce4a034f738915af062eb69fb6","98898837d84743619f4e5346174f3b42","dfe1fa6f62fb4c5da30dd645213d3ab4"],"click_id":{},"doc_encoding":"UTF-8","doc_host":"fw.sw.local","doc_path":"/68930-natuli-porod-naturalny.html","doc_search":"","ech_id":"nikhkxnvcqb9wf7q_ur","ech_processed":0,"event_type":"view_item","eventn_ctx_event_id":"4012e69b-257c-4aab-b2fe-e621f0cb2efd","ids":{"ga":"GA1.1.646209339.1662539324"},"local_tz_offset":-120,"manufacturerId":"a0392837039b468a9d819779c73b0d5f","onStock":true,"page_title":"Natuli, Poród naturalny","price":"45","priceRange":"0-50","productId":"4e29d557497440068a2f2eb7139ee801","referer":"http://fw.sw.local/68-poradniki-dla-rodzicow","screen_resolution":"1792x1120","source_ip":"172.29.0.1","src":"jitsu","url":"http://fw.sw.local/68930-natuli-porod-naturalny.html","user":{"anonymous_id":"rqm0xasw6g","hashed_anonymous_id":"ffb744599097e7c1a4e0b15d7a292f3d"},"user_agent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/15.6.1 Safari/605.1.15","user_language":"en-GB","utc_time":"2022-09-07T08:34:44.367000Z","utm":{},"vp_size":"1744x534"},"error":"failed to apply builtin javascript transform: json: unsupported type: map[interface {}]interface {}","event_id":"4012e69b-257c-4aab-b2fe-e621f0cb2efd","RecognizedEvent":false}
```

test results:

```
/usr/local/bin/yarn test
yarn run v1.22.4
$ jitsu-cli extension test
[info ] - 🛂 Running tests for /Users/lutek/Projects/jitsu/jitsu-harness-destination
[info ] - ℹ️ Found tsconfig.json, typescript will be enabled
 PASS  __test__/destination.test.ts
  ✓ update_item (2 ms)
  ✓ delete_item
  ✓ view_item event (1 ms)
  ✓ view_item event no user (1 ms)
  ✓ view_item event no productId (1 ms)
  ✓ add_to_cart event
  ✓ add_to_cart2 non existing event (1 ms)
  ✓ buy event (1 ms)

Test Suites: 1 passed, 1 total
Tests:       8 passed, 8 total
Snapshots:   0 total
Time:        2.081 s
Ran all test suites.
[info ] - ✨ Done
✨  Done in 4.47s.

```

exec results:
```
$ jitsu-cli extension exec --file examples/test-events-view_item.json -c examples/test-config.json
[info ] - ⌛️ Loading extension (don't forget to build it before running exec!). Source: /Users/lutek/Projects/jitsu/jitsu-harness-destination/dist/jitsu-harness-destination.js
[info ] - 🛂 Executing tests destination on /Users/lutek/Projects/jitsu/jitsu-harness-destination
[info ] - Validating configuration:
    {
      "engine_url": "https://rec.ecommercebox.io",
      "item_id": "productId",
      "group_view_item_event_types": [
        {
          "view_item": "productId"
        },
        {
          "view_item_p_03fa174193cc43ea9d58458f1bdb8193": "p_03fa174193cc43ea9d58458f1bdb8193"
        },
        {
          "view_item_manufacturer": "manufacturerId"
        },
        {
          "view_item_price_range": "priceRange"
        }
      ],
      "group_buy_event_types": [
        {
          "buy": "items.*.id"
        },
        {
          "buy_price_range": "items.*.priceRange"
        },
        {
          "buy_manufacturer": "items.*.manufacturerId"
        },
        {
          "buy_p_03fa174193cc43ea9d58458f1bdb8193": "items.*.properties.p_03fa174193cc43ea9d58458f1bdb8193"
        },
        {
          "buy_p_26ab5f36aa5e4aa694e860617f575390": "items.*.properties.p_26ab5f36aa5e4aa694e860617f575390"
        }
      ],
      "event_types": [
        {
          "update_item": "productId"
        },
        {
          "add_to_cart": "productId"
        },
        {
          "add_to_wishlist": "productId"
        }
      ]
    }

[info ] - ✅ Configuration is valid!
[info ] - 🏃 Running destination plugin on 1 events
[info ] - ✅ Event emitted 8 messages. Event JSON: {"_timestamp":"2022-08-31T14:36:23.162639Z","api_key":"xxx","available":true,...
[info ] -     POST https://rec.ecommercebox.io/engines/test_ur/queries
[info ] -     Body:
        {
          "event": "view_item",
          "entityType": "user",
          "entityId": "f660ab912ec121d1b1e928a0bb4bc61b15f5ad44d5efdc4e1c92a25e99b8e44a",
          "targetEntityType": "item",
          "targetEntityId": "2065f844cb5c4b6a850025764bbf7676",
          "eventTime": "2022-08-31T14:36:23.162Z"
        }
[info ] -     POST https://rec.ecommercebox.io/engines/test_ur/queries
[info ] -     Body:
        {
          "event": "view_item_p_03fa174193cc43ea9d58458f1bdb8193",
          "entityType": "user",
          "entityId": "f660ab912ec121d1b1e928a0bb4bc61b15f5ad44d5efdc4e1c92a25e99b8e44a",
          "targetEntityType": "item",
          "targetEntityId": "2f8be30746ca409f864eb65cc1beb577",
          "eventTime": "2022-08-31T14:36:23.162Z"
        }
[info ] -     POST https://rec.ecommercebox.io/engines/test_ur/queries
[info ] -     Body:
        {
          "event": "view_item_p_03fa174193cc43ea9d58458f1bdb8193",
          "entityType": "user",
          "entityId": "f660ab912ec121d1b1e928a0bb4bc61b15f5ad44d5efdc4e1c92a25e99b8e44a",
          "targetEntityType": "item",
          "targetEntityId": "37058b9bc5d643bab8ac5e7b70f4f834",
          "eventTime": "2022-08-31T14:36:23.162Z"
        }
[info ] -     POST https://rec.ecommercebox.io/engines/test_ur/queries
[info ] -     Body:
        {
          "event": "view_item_p_03fa174193cc43ea9d58458f1bdb8193",
          "entityType": "user",
          "entityId": "f660ab912ec121d1b1e928a0bb4bc61b15f5ad44d5efdc4e1c92a25e99b8e44a",
          "targetEntityType": "item",
          "targetEntityId": "b51b1eca0de04919b8d6bd38ba18b02f",
          "eventTime": "2022-08-31T14:36:23.162Z"
        }
[info ] -     POST https://rec.ecommercebox.io/engines/test_ur/queries
[info ] -     Body:
        {
          "event": "view_item_p_03fa174193cc43ea9d58458f1bdb8193",
          "entityType": "user",
          "entityId": "f660ab912ec121d1b1e928a0bb4bc61b15f5ad44d5efdc4e1c92a25e99b8e44a",
          "targetEntityType": "item",
          "targetEntityId": "cf3daddb39734dd08f9de9ae52b6a04f",
          "eventTime": "2022-08-31T14:36:23.162Z"
        }
[info ] -     POST https://rec.ecommercebox.io/engines/test_ur/queries
[info ] -     Body:
        {
          "event": "view_item_p_03fa174193cc43ea9d58458f1bdb8193",
          "entityType": "user",
          "entityId": "f660ab912ec121d1b1e928a0bb4bc61b15f5ad44d5efdc4e1c92a25e99b8e44a",
          "targetEntityType": "item",
          "targetEntityId": "d9478b66d93e42aca2eab87f61c36dd8",
          "eventTime": "2022-08-31T14:36:23.162Z"
        }
[info ] -     POST https://rec.ecommercebox.io/engines/test_ur/queries
[info ] -     Body:
        {
          "event": "view_item_manufacturer",
          "entityType": "user",
          "entityId": "f660ab912ec121d1b1e928a0bb4bc61b15f5ad44d5efdc4e1c92a25e99b8e44a",
          "targetEntityType": "item",
          "targetEntityId": "5ae2d827f0524c6389dbaef87ccbd73c",
          "eventTime": "2022-08-31T14:36:23.162Z"
        }
[info ] -     POST https://rec.ecommercebox.io/engines/test_ur/queries
[info ] -     Body:
        {
          "event": "view_item_price_range",
          "entityType": "user",
          "entityId": "f660ab912ec121d1b1e928a0bb4bc61b15f5ad44d5efdc4e1c92a25e99b8e44a",
          "targetEntityType": "item",
          "targetEntityId": "100-150",
          "eventTime": "2022-08-31T14:36:23.162Z"
        }
[info ] - ✨ Done
✨  Done in 4.28s.
```

server config
```yaml
server:
  port: 8001
  disable_welcome_page: true
  name: e1-ecommercebox 

log:
  path: /home/eventnative/data/logs
  rotation_min: 5
  show_in_server: true

api_keys:
  - id: fw_local_events
    client_secret: js.e1af3113
    server_secret: s2s.68b6c3c4
    origins:
      - 'fw.local'
      - 'fw.sw.local'

destinations:
  mysql_webevents:
    type: mysql
    mode: stream #Optional. Default value is 'batch'
    only_tokens: [ 'fw_local_events' ]
    datasource:
      schema: webevents #'public' is default value
      host: mysql
      port: 3306
      db: ps17
      username: root
      password: root
      parameters:
        tls: 'false'
    data_layout:
      table_name_template: 'rec_events_{{._timestamp.Format "2006"}}'
    users_recognition: #Optional
      enabled: false #set false for disabling
  rec_ecommercebox:
    only_tokens: [ 'fw_local_events' ]
    type: npm
    package: /home/eventnative/data/plugins/jitsu-harness-destination.tgz
    mode: stream
    config:
      engine_url: "https://rec.ecommercebox.io"
      item_id: "productId"
      group_view_item_event_types: [
        {
          "view_item": "productId"
        },
        {
          "view_item_p_03fa174193cc43ea9d58458f1bdb8193": "p_03fa174193cc43ea9d58458f1bdb8193"
        },
        {
          "view_item_manufacturer": "manufacturerId"
        },
        {
          "view_item_price_range": "priceRange"
        }
      ]
      group_buy_event_types: [
        {
          "buy": "items.*.id"
        },
        {
          "buy_price_range": "items.*.priceRange"
        },
        {
          "buy_manufacturer": "items.*.manufacturerId"
        },
        {
          "buy_p_03fa174193cc43ea9d58458f1bdb8193": "items.*.properties.p_03fa174193cc43ea9d58458f1bdb8193"
        },
        {
          "buy_p_26ab5f36aa5e4aa694e860617f575390": "items.*.properties.p_26ab5f36aa5e4aa694e860617f575390"
        }
      ]
      event_types: [
        {
          "update_item": "productId"
        },
        {
          "add_to_cart": "productId"
        },
        {
          "add_to_wishlist": "productId"
        }
      ]

users_recognition: #Optional
  enabled: true #set false for disabling
  anonymous_id_node: /user/anonymous_id
  identification_nodes:
    - /user/email

meta:
  storage:
    redis:
      host: redis
      port: 6379
```

plugin with examples:
[jitsu-harness-destination.tgz.zip](https://github.com/jitsucom/jitsu/files/9504454/jitsu-harness-destination.tgz.zip)

**Comment by absorbb** [comment]
@lutek 
That might be caused by the way how you put event types settings into yaml config.
Please try the following way:
```yaml
 rec_ecommercebox:
    only_tokens:
      - fw_local_events
    type: npm
    package: /home/eventnative/data/plugins/jitsu-harness-destination.tgz
    mode: stream
    config:
      engine_url: https://rec.ecommercebox.io
      item_id: productId
      event_types:
        - update_item: productId
        - add_to_cart: productId
        - add_to_wishlist: productId
      group_buy_event_types:
        - buy: items.*.id
        - buy_price_range: items.*.priceRange
        - buy_manufacturer: items.*.manufacturerId
        - buy_p_03fa174193cc43ea9d58458f1bdb8193: items.*.properties.p_03fa174193cc43ea9d58458f1bdb8193
        - buy_p_26ab5f36aa5e4aa694e860617f575390: items.*.properties.p_26ab5f36aa5e4aa694e860617f575390
      group_view_item_event_types:
        - view_item: productId
        - view_item_p_03fa174193cc43ea9d58458f1bdb8193: p_03fa174193cc43ea9d58458f1bdb8193
        - view_item_manufacturer: manufacturerId
        - view_item_price_range: priceRange
```

It works for me

**Comment by absorbb** [comment]
also i would suggest using more simple structure like:
```yaml
event_types:
  update_item: productId
  add_to_cart: productId
  add_to_wishlist: productId
```
but that will probably require small changes in the code

**Comment by lutek** [comment]
@absorbb i change config but still got the same error :/ 
is there in jitsu server debug log to see what happens?

> @lutek That might be caused by the way how you put event types settings into yaml config. Please try the following way:
> 
> 
> It works for me

**Comment by absorbb** [comment]
@lutek i've reproduced the issue. working on the fix

**Comment by lutek** [comment]
@absorbb I'm guessing it works, but I'm running out of ideas what the problem is.

![image](https://user-images.githubusercontent.com/846598/188907777-e232961c-d81b-4307-800d-f61f4cd28229.png)

**Comment by absorbb** [comment]
@lutek  problem is fixed in `beta` docker image tag. you can try it.
Will be released as a hotfix release in the next few days.

**Comment by lutek** [comment]
@absorbb on beta is working. thank you!

**Comment by absorbb** [comment]
Also fixed in: [1.42.8](https://github.com/jitsucom/jitsu/releases/tag/v1.42.8)
